napari/utils/transforms/transform_utils.py

```diff
@@ -29,51 +29,220 @@ def compose_linear_matrix(rotate, scale, shear) -> np.array:
+    ndim : int
+        The desired dimensionality of the output transform component.
+
+    Returns
+    -------
+    np.ndarray
+        The rotate component as a 2D numpy array of size ndim.
+    """
+    full_rotate_mat = np.eye(ndim)
+    if rotate is not None:
+        rotate_mat = _make_rotate_mat(rotate)
+        rotate_mat_ndim = rotate_mat.shape[0]
+        full_rotate_mat[-rotate_mat_ndim:, -rotate_mat_ndim:] = rotate_mat
+    return full_rotate_mat
+
+
+def _make_rotate_mat(rotate):
     if np.isscalar(rotate):
-        # If a scalar is passed assume it is a single rotation angle
-        # for a 2D rotation
-        theta = np.deg2rad(rotate)
-        rotate_mat = np.array(
-            [[np.cos(theta), -np.sin(theta)], [np.sin(theta), np.cos(theta)]]
-        )
+        return _make_2d_rotation(rotate)
     elif np.array(rotate).ndim == 1 and len(rotate) == 3:
-        # If a 3-tuple is passed assume it is three rotation angles for
-        # a roll, pitch, and yaw for a 3D rotation. For more details see
-        # https://en.wikipedia.org/wiki/Rotation_matrix#General_rotations
-        alpha = np.deg2rad(rotate[0])
-        beta = np.deg2rad(rotate[1])
-        gamma = np.deg2rad(rotate[2])
-        R_alpha = np.array(
-            [
-                [np.cos(alpha), np.sin(alpha), 0],
-                [-np.sin(alpha), np.cos(alpha), 0],
-                [0, 0, 1],
-            ]
-        )
-        R_beta = np.array(
-            [
-                [np.cos(beta), 0, np.sin(beta)],
-                [0, 1, 0],
-                [-np.sin(beta), 0, np.cos(beta)],
-            ]
-        )
-        R_gamma = np.array(
-            [
-                [1, 0, 0],
-                [0, np.cos(gamma), -np.sin(gamma)],
-                [0, np.sin(gamma), np.cos(gamma)],
-            ]
-        )
-        rotate_mat = R_alpha @ R_beta @ R_gamma
-    else:
-        # Otherwise assume a full nD rotation matrix has been passed
-        rotate_mat = np.array(rotate)
-    n_rotate = rotate_mat.shape[0]
+        return _make_3d_rotation(*rotate)
+    return np.array(rotate)
+
+
+def _make_2d_rotation(theta_degrees):
+    """Makes a 2D rotation matrix from an angle in degrees."""
+    cos_theta, sin_theta = _cos_sin_degrees(theta_degrees)
+    return np.array([[cos_theta, -sin_theta], [sin_theta, cos_theta]])
 
-    # Convert a scale vector to an nD diagonal matrix
-    scale_mat = np.diag(scale)
-    n_scale = scale_mat.shape[0]
 
+def _make_3d_rotation(alpha_degrees, beta_degrees, gamma_degrees):
+    """Makes a 3D rotation matrix from roll, pitch, and yaw in degrees.
+
+    For more details, see: https://en.wikipedia.org/wiki/Rotation_matrix#General_rotations
+    """
+    cos_alpha, sin_alpha = _cos_sin_degrees(alpha_degrees)
+    R_alpha = np.array(
+        [
+            [cos_alpha, -sin_alpha, 0],
+            [sin_alpha, cos_alpha, 0],
+            [0, 0, 1],
+        ]
+    )
+
+    cos_beta, sin_beta = _cos_sin_degrees(beta_degrees)
```

Comment:
Added some trig pre-calculations as suggested by @alisterburt in another related PR: https://github.com/napari/napari/pull/2793#discussion_r641840161

Reply:
i like it